src/librbd/Operations.cc

```diff
@@ -1355,7 +1355,9 @@ int Operations<I>::metadata_set(const std::string &key,
   string start = m_image_ctx.METADATA_CONF_PREFIX;
   size_t conf_prefix_len = start.size();
 
-  if (key.size() > conf_prefix_len && !key.compare(0, conf_prefix_len, start)) {
+  bool config_key = (key.size() > conf_prefix_len &&
+                     !key.compare(0, conf_prefix_len, start));
+  if (config_key) {
```

Comment:
super nit: may be introduce a helper function in ImageContex returning if it is config key and its name (subkey)? So `METADATA_CONF_PREFIX` would not need to be exposed and it could be reused in `metadata_remove`.